server/public/shared/httpservice/httpservice.go

```diff
@@ -88,31 +89,31 @@ func (h *HTTPServiceImpl) MakeTransport(trustURLs bool) *MattermostTransport {
 		return false
 	}
 
-	allowIP := func(ip net.IP) bool {
+	allowIP := func(ip net.IP) error {
 		reservedIP := IsReservedIP(ip)
-		ownIP, err := IsOwnIP(ip)
 
-		// If there is an error getting the self-assigned IPs, default to the secure option
+		ownIP, err := IsOwnIP(ip)
 		if err != nil {
-			return false
+			// If there is an error getting the self-assigned IPs, default to the secure option
+			return fmt.Errorf("unable to determine if IP is own IP: %w", err)
 		}
 
 		// If it's not a reserved IP and it's not self-assigned IP, accept the IP
 		if !reservedIP && !ownIP {
-			return true
-		}
-
-		if h.configService.Config().ServiceSettings.AllowedUntrustedInternalConnections == nil {
-			return false
+			return nil
 		}
 
 		// In the case it's the self-assigned IP, enforce that it needs to be explicitly added to the AllowedUntrustedInternalConnections
 		for _, allowed := range strings.FieldsFunc(*h.configService.Config().ServiceSettings.AllowedUntrustedInternalConnections, splitFields) {
```

Comment:
Could this panic if `h.configService.Config().ServiceSettings.AllowedUntrustedInternalConnections == nil`?

Reply:
Good point, the code now catches that using `model.SafeDereference`.